### PR TITLE
docs: adding PESQ example to gallery

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

adding another sample

<details>
  <summary>Before submitting</summary>

- [x] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2755.org.readthedocs.build/en/2755/

### CODE REVIEW

**Comment by Borda** [comment]
> @Borda what is missing here?

I was not sure if the story is rich enough

File: examples/audio/pesq.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
+"""
+PESQ Metric Calculation for Speech Enhancement
```
Comment (SkafteNicki) [inline examples/audio/pesq.py:2]:
```suggestion
Evaluating Speech Quality with PESQ metric
```

File: examples/audio/pesq.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+==============================================
+
+In this notebook, we will calculate the Perceptual Evaluation of Speech Quality (PESQ) score
+to assess the improvement in speech quality after applying a basic noise reduction technique.
```
Comment (SkafteNicki) [inline examples/audio/pesq.py:6]:
```suggestion
This notebook will guide you through calculating the Perceptual Evaluation of Speech Quality (PESQ) score, a key metric in assessing how effective noise reduction and enhancement techniques are in improving speech quality. PESQ is widely adopted in industries such as telecommunications, VoIP, and audio processing. It provides an objective way to measure the perceived quality of speech signals from a human listener's perspective.
```

File: examples/audio/pesq.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+PESQ is widely used in speech enhancement, telecommunications, and VoIP to evaluate the
+perceived quality of speech signals.
```
Comment (SkafteNicki) [inline examples/audio/pesq.py:9]: Imagine you’re on a noisy street, trying to have a phone call. The technology behind the scenes aims to clean up your voice and make it sound clearer on the other end. But how do engineers measure that improvement? This is where PESQ comes in. In this notebook, we will simulate a similar scenario, applying a simple noise reduction technique and using the PESQ score to evaluate how much the speech quality improves.

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**